geth/api/backend.go

```diff
@@ -77,120 +80,145 @@ func (m *StatusBackend) IsNodeRunning() bool {
 }
 
 // StartNode start Status node, fails if node is already started
-func (m *StatusBackend) StartNode(config *params.NodeConfig) (<-chan struct{}, error) {
+func (m *StatusBackend) StartNode(config *params.NodeConfig) error {
 	m.Lock()
 	defer m.Unlock()
+	return m.startNode(config)
+}
 
-	if m.nodeReady != nil {
-		return nil, node.ErrNodeExists
-	}
-
-	nodeStarted, err := m.nodeManager.StartNode(config)
-	if err != nil {
-		return nil, err
+func (m *StatusBackend) startNode(config *params.NodeConfig) error {
+	if err := m.nodeManager.StartNode(config); err != nil {
+		switch err.(type) {
+		case node.RPCClientError:
+			signal.Send(signal.Envelope{
+				Type: signal.EventNodeCrashed,
+				Event: signal.NodeCrashEvent{
+					Error: node.ErrRPCClient.Error(),
+				},
+			})
+		case node.EthNodeError:
+			signal.Send(signal.Envelope{
+				Type: signal.EventNodeCrashed,
+				Event: signal.NodeCrashEvent{
+					Error: fmt.Errorf("%v: %v", node.ErrNodeStartFailure, err).Error(),
+				},
+			})
+		default:
+			signal.Send(signal.Envelope{
+				Type: signal.EventNodeCrashed,
+				Event: signal.NodeCrashEvent{
+					Error: fmt.Errorf("Node creation failed: %v", err).Error(),
+				},
+			})
+		}
+
+		return err
 	}
-
-	m.nodeReady = make(chan struct{}, 1)
-	go m.onNodeStart(nodeStarted, m.nodeReady) // waits on nodeStarted, writes to backendReady
-
-	return m.nodeReady, err
+	signal.Send(signal.Envelope{
+		Type:  signal.EventNodeStarted,
+		Event: struct{}{},
```

Comment:
I remember seeing `type empty struct{}` somewhere. Should we use a similar pattern here?

Reply:
i wonder why this field can't be null

Reply:
@dshulyak I see that `signal.Send` encodes the `Envelope` to json before calling `C.StatusServiceSignalEvent` with it. The `event` value would be `null` instead of `{}`, so it depends if it make sense there.

Reply:
thanks @pilu, i saw it. i will to reach consumers of this api (status-react) and see if it makes sense for them if this field will be omitted in case if it is empty